module/VuFind/src/VuFind/Controller/OverdriveController.php

```diff
@@ -0,0 +1,211 @@
+<?php
+/**
+ * Overdrive Controller
+ *
+ * PHP version 7
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Brent Palmer <brent-palmer@ipcl.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org Main Site
+ */
+ namespace VuFind\Controller;
+ use Zend\ServiceManager\ServiceLocatorInterface;
+ use Zend\Log\LoggerAwareInterface;
+ 
+/**
+ * Overdrive Controller supports action for Overdrive Integration
+ *
+ * @category VuFind
+ * @package  Controller
+ * @author   Brent Palmer <brent-palmer@ipcl.org>
+ * @license  http://opensource.org/licenses/gpl-2.0.php GNU General Public License
+ * @link     https://vufind.org/wiki/development:plugins:controllers Wiki
+ */ 
+class OverdriveController extends AbstractBase implements LoggerAwareInterface
+{
+    use \VuFind\Log\LoggerAwareTrait {
+        logError as error;
+    }
+    
+    protected $connector;
+    
+    /**
+     * Constructor
+     *
+     * @param ServiceLocatorInterface $sm Service locator
+     */
+    public function __construct(ServiceLocatorInterface $sm)
+    {
+        $this->setLogger($sm->get('VuFind\Logger'));
```

Comment:
You don't need to manually set the logger here -- by virtue of the LoggerAwareInterface being on the class, the initializer code should automatically inject the logger for you.

Reply:
I think I got this from you.  this controller still needs it I guess:
See https://sourceforge.net/p/vufind/mailman/vufind-general/?viewmonth=201803&viewday=13

I just tried commenting it out to see but the logging disappeard.

Reply:
It's dawning on me that I missed a step when I did the PR.  I forgot to get the latest upstream changes and merge them in.  I'll work on that before I continue.

Reply:
Whoops -- slip of the memory. Sorry about that! :-)